### PR TITLE
[next] overlay/15fcos: upgrade bootloader on aarch64 at boot

### PR DESCRIPTION
**Pull request by bgilbert**
On aarch64, kernel 6.2 won't boot with older versions of GRUB.  In preparation for switching to the new kernel, add a systemd service that uses bootupd to update the bootloader on aarch64 systems.

Revert this after the next barrier release.

For https://github.com/coreos/fedora-coreos-tracker/issues/1441.  Not heavily tested.

### CODE REVIEW

**Comment by bgilbert** [comment]
Downgrade detector detected downgrades.

**Comment by dustymabe** [comment]
This seems to be working in a preliminary test for me. Want to open a similar PR up against `testing-devel` since we'll need this in the other streams too?

I'll try to do some more testing tomorrow.

**Comment by bgilbert** [comment]
`testing-devel` in #2308.

**Comment by dustymabe** [comment]
I tested this with a new upgrade script I wrote.. Here is some selected output:

```
[builder@coreos-aarch64-builder fcos]$ tail -f  tmp/kola/ext.config.upgrade/*/journal.txt | grep --color -i 'ok reached version'                                                   
Mar 20 02:25:12.584096 kola-runext-test.sh[2085]: ok Reached version: 34.20210904.1.0                                                                                              
Mar 20 02:26:37.107738 kola-runext-test.sh[3538]: ok Reached version: 35.20210924.1.0                                                                                              
Mar 20 02:28:08.509482 kola-runext-test.sh[1668]: ok Reached version: 36.20220418.1.0                                                                                              
Mar 20 02:28:41.242555 kola-runext-test.sh[1649]: ok Reached version: 36.20220906.1.0                                                                                                                                                                                                                                                                     
Mar 20 02:29:33.635979 kola-runext-test.sh[1635]: ok Reached version: 37.20221111.1.0                                                                                              
Mar 20 02:30:16.657854 kola-runext-test.sh[1469]: ok Reached version: 37.20230303.1.0                                                                                              
Mar 20 02:31:17.440458 kola-runext-test.sh[1297]: ok Reached version: 37.20230320.dev.0                                                                                            
Mar 20 02:32:14.601624 kola-runext-test.sh[1264]: ok Reached version: 38.20230310.1.0
```

The `37.20230320.dev.0` is a local build that includes this PR.

The `bootupctl update` looks like:

```
Mar 20 02:31:04.636253 systemd[1]: Starting coreos-bootupctl-update-aarch64.service - Update aarch64 Bootloader...
Mar 20 02:31:05.087429 bootupctl[740]: Updated EFI: grub2-efi-aa64-1:2.06-88.fc37.aarch64,shim-aa64-15.6-2.aarch64
Mar 20 02:31:05.088124 systemd[1]: Finished coreos-bootupctl-update-aarch64.service - Update aarch64 Bootloader.
```

File: overlay.d/15fcos/usr/lib/systemd/system/coreos-bootupctl-update-aarch64.service [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Remove after the next barrier release
+# https://github.com/coreos/fedora-coreos-tracker/issues/1441
```
Comment (jlebon) [inline overlay.d/15fcos/usr/lib/systemd/system/coreos-bootupctl-update-aarch64.service:2]: Minor: could make this a `Documentation` key.

**Review by jlebon — APPROVED** [review]
Nice and simple!

**Comment by jlebon** [comment]
Also tested this manually locally with success starting from the very first aarch64 next image. Rebooting into f38 after the bootloader autoupdate worked.

**Pull request merged**